### ISSUE TITLE
Should we move to UTF-8 Only?

### ISSUE THREAD

**Issue by spekary** [qcubed/qcubed#983]
I am struggling with a new feature for plugins, and I want to use JSON.stringify in javascript, and decode it using PHP's json_decode. The problem is that json_decode is UTF-8 only, so I suspect it will require that the web page be UTF-8 encoded.

Is there any reason that requiring UTF-8 encoding on all qcubed web pages from here on out might be a bad thing?

**Comment by olegabr** [comment]
For me UTF-8 is ok, but any limitation should be a cost for something really good.
Can you please describe this new feature in more details?

**Comment by spekary** [comment]
Never mind. The issue I was struggling with is moving data from qcubed.js to PHP. I have discovered that stringifying data with $j.param() or JSON.stringify() will encode data in internal browser format, which for modern browsers is going to be UTF. This is regardless of the character encoding specified in the HTML. That means when you try to unpack on the PHP side using parse_str or json_decode, you will get UTF-8, and trying to send that to a QControl will result in bad data if the application is not set to UTF-8. 

I have found a way to detect the problem and fix it on the PHP side, and will try to post a fix soon, including some testing code to double-check. The fix should accommodate any PHP supported encoding.
